ext/standard/basic_functions.stub.php

```diff
@@ -234,21 +222,16 @@ function array_sum(array $arg): int|float {}
 
 function array_product(array $arg): int|float {}
 
-/** @return mixed */
-function array_reduce(array $arg, callable $callback, $initial = null) {}
+function array_reduce(array $arg, callable $callback, $initial = null): mixed {}
 
 function array_filter(array $arg, ?callable $callback = null, int $use_keys = 0): array {}
 
 function array_map(?callable $callback, array $arr1, array ...$arrays): array {}
 
-/** @param mixed $key */
-function array_key_exists($key, array $search): bool {}
+function array_key_exists(mixed $key, array $search): bool {}
```

Comment:
Not sure on this one, but might be better to make this an `int|string` doc only type?

Reply:
Yes, I am also a little bit unsure. The reason why I did so is that I felt that it might be useful to indicate that you can provide any type (with the exception of objects) as the first argument.

But then I provided a reason why you are right. :) We shouldn't annotate parameters with `mixed` when they don't support all the types (since it can lead to type errors), while it's not that big of a deal in case of return types.